igor-web/src/main/java/com/netflix/spinnaker/igor/codebuild/AwsCodeBuildRequestHandler.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2020 Amazon.com, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.igor.codebuild;
+
+import com.amazonaws.Request;
+import com.amazonaws.Response;
+import com.amazonaws.handlers.RequestHandler2;
+import com.amazonaws.services.codebuild.model.AWSCodeBuildException;
+import com.netflix.spinnaker.igor.exceptions.BuildJobError;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class AwsCodeBuildRequestHandler extends RequestHandler2 {
+  @Override
+  public void afterError(Request<?> request, Response<?> response, Exception e) {
+    if (e instanceof AWSCodeBuildException) {
```

Comment:
Would this include ThrottlingException?  Should API rate throttling be a build job error or a runtime exception?

Reply:
ThrottlingException does not inherit from AWSCodeBuildException so it will be a RuntimeException

Reply:
I'm not sure what the implications of picking build job error vs runtime exception are for how Spinnaker behaves when these are returned; I'm not super familiar with Igor.  Why should ThrottlingException be runtime error but AccountLimitExceededException be a build error?

Reply:
BuildJobError is returned as 400 while RuntimeException will become an InternalError and return 500. I think throttling exception should probably return 400 instead of 500. I will categorize it as BuildJobError as well.

Reply:
You could also look at the exception error type:
https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/AmazonServiceException.ErrorType.html